crates/mun_hir/src/item_tree.rs

```diff
@@ -302,11 +303,22 @@ pub struct Function {
     pub visibility: RawVisibilityId,
     pub is_extern: bool,
     pub types: TypeRefMap,
-    pub params: Box<[LocalTypeRefId]>,
+    pub params: IdRange<Param>,
     pub ret_type: LocalTypeRefId,
     pub ast_id: FileAstId<ast::FunctionDef>,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Param {
+    pub type_ref: LocalTypeRefId,
+    pub ast_id: ParamAstId,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ParamAstId {
```

Comment:
What other variants might this enum contain?

Reply:
self! 🙂